Cython/Compiler/Parsing.py

```diff
@@ -1082,7 +1082,7 @@ def p_f_string(s, unicode_value, pos, is_raw):
                     values.append(ExprNodes.UnicodeNode(pos, value=builder.getstring()))
                     builder = StringEncoding.UnicodeLiteralBuilder()
                 next_start, expr_node = p_f_string_expr(s, unicode_value, pos, next_start, is_raw)
-                values.append(expr_node)
+                values.extend(expr_node)
```

Comment:
```suggestion
                next_start, expr_nodes = p_f_string_expr(s, unicode_value, pos, next_start, is_raw)
                values.extend(expr_nodes)
```

Other than this tiny comment I think this makes sense - it's does look simpler to do this in the parser.

Reply:
Yep, this looks like a much nicer way to do it. Thanks for the suggestion :)

PS: Fixed